components-compose/src/main/java/uk/gov/hmrc/components/compose/organism/summaryrow/SummaryRowView.kt

```diff
@@ -0,0 +1,142 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.hmrc.components.compose.organism.summaryrow
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.ripple.LocalRippleTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.semantics.focused
+import androidx.compose.ui.semantics.role
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.style.TextOverflow
+import uk.gov.hmrc.components.compose.R
+import uk.gov.hmrc.components.compose.ui.theme.HmrcRippleTheme
+import uk.gov.hmrc.components.compose.ui.theme.HmrcTheme.dimensions
+import uk.gov.hmrc.components.compose.ui.theme.HmrcTheme.typography
+
+object SummaryRowView {
+
+    @Composable
+    operator fun invoke(
+        modifier: Modifier = Modifier,
+        titleText: String,
+        isBoldTitleTextAppearance: Boolean = true,
+        titleMaxLines: Int = -1,
+        icon: Painter = painterResource(id = R.drawable.components_ic_chevron_right),
+        chevronContentDescription: String = "",
+        accessibilityMessage: String? = null,
+        rows: List<@Composable () -> Unit>,
+        readerTrait: ReaderTrait = ReaderTrait.READER_TRAIT_INFO,
+        onSummaryRowClicked: (() -> Unit)? = null
+    ) {
+        if (onSummaryRowClicked != null) {
+            CompositionLocalProvider(LocalRippleTheme provides HmrcRippleTheme) {
+                Row(
+                    modifier = Modifier.clickable { onSummaryRowClicked.invoke() },
+                    verticalAlignment = Alignment.CenterVertically
+                ) {
+                    SummaryRow(
+                        modifier = modifier.then(
+                            Modifier
```

Comment:
I think we probably want the modifier that gets passed through from the consumer to apply to the outer row rather than the inner SummaryRow.. 
```suggestion
                Row(
                    modifier = modifier.clickable { onSummaryRowClicked.invoke() },
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    SummaryRow(
                        modifier = Modifier
```